earthly/python/Earthfile

```diff
@@ -1,42 +1,105 @@
 # Common Python UDCs and Builders.
 VERSION --global-cache --use-function-keyword 0.7
 
+# cspell: words libgcc ruff
+
 python-base:
     FROM python:3.12-alpine3.19
 
-    # Install extra packages we will need to support plugins.
-    RUN apk add --no-cache \    
+    WORKDIR /root
+
+    # Install necessary packages 
+    RUN apk add --no-cache \
+        bash \
         curl \
         libffi-dev \
         gcc \
         musl-dev \
-        libgcc
+        libgcc \
+        zlib-dev \
+        jpeg-dev \
+        git
 
     # Poetry Installation directory.
     # Gives poetry and our poetry project a reliable location.
-    WORKDIR /poetry
-    ENV POETRY_HOME=/poetry
+    ENV POETRY_HOME=/root/poetry
     ENV PATH=$POETRY_HOME/bin:$PATH
 
     # Stop python code running in the container saving pointless .pyc files.
     ENV PYTHONDONTWRITEBYTECODE=1
 
-    # Install Poetry to manage python.
+    # Install Poetry using the recommended installer.
     RUN curl -sSL https://install.python-poetry.org | python3 -
-    # Prevent warning when running on machines with more than 6 cores.
-    #   "Connection pool is full, discarding connection: pypi.org. Connection pool size: 10"
+
+    # Adjust Poetry's configuration to prevent connection pool warnings.
     RUN poetry config installer.max-workers 10
 
-# Common python builder setup.
-# Parameters:
-#   opts      : Options passed to the `poetry install` command.
+    # Install ruff for linting.
+    RUN pip3 install ruff
+
+    # Universal build scripts we will always need and are not target dependent.
+    COPY --dir scripts /scripts
+    # Copy our common scripts so we can use them inside the container.
+    DO ../../utilities/scripts+ADD_BASH_SCRIPTS
+    DO ../../utilities/scripts+ADD_PYTHON_SCRIPTS
+
+    SAVE ARTIFACT /scripts /scripts
+
 BUILDER:
     FUNCTION
-    ARG opts
 
-    # Copy our dependencies.
-    COPY pyproject.toml poetry.lock Readme.md .
+    ARG poetry_lock=./poetry.lock
+    ARG pyproject_toml=./pyproject.toml
+
+    FROM +python-base
+    WORKDIR /build
+
+    COPY --dir $poetry_lock .
+    COPY --dir $pyproject_toml .
+
+CHECK:
+    FUNCTION
+
+    FROM +python-base
+    WORKDIR /build
+
+    # Copy the necessary files into the container
+    COPY pyproject.toml .
+    COPY poetry.lock .
+
+    # Ensure scripts have executable permissions
+    RUN chmod +x /scripts/*.py
+
+    # Debugging step to verify environment setup
+    RUN ls -l /scripts && which python3 && echo $PATH
+
+    # Execute the check script
+    RUN /scripts/std_checks.py
+
+FORMAT:
+    FUNCTION
+
+    # Can not run format LOCALLY as it changes permissions of files to root:root
+    RUN --no-cache printf "%s\n%s\n%s\n" \
```

Comment:
is this mean using `ruff format`?

Reply:
yes, automatically adjusting the code to adhere to ruff's linting protocols

Reply:
decided to delete FORMAT commit 98485de21c7132e2bdb4c0c02ba4797d54783e5f